falcon/responders.py

```diff
@@ -38,11 +45,15 @@ def create_method_not_allowed(allowed_methods):
             returned in the Allow header.
 
     """
-    def method_not_allowed(req, resp, **kwargs):
-        """Raise 405 HTTPMethodNotAllowed error"""
-        raise HTTPMethodNotAllowed(allowed_methods)
+    partial_method_not_allowed = partial(method_not_allowed, allowed_methods)
```

Comment:
[observation] Although partials are less performant than closures, in this case I think it's OK since these code paths are less latency-sensitive.